ui/nuxeo-document-preview.js

```diff
@@ -283,7 +283,12 @@ import './viewers/nuxeo-video-viewer.js';
       ) {
         const conversions = [];
         this.document.properties['vid:transcodedVideos'].forEach((conversion) => {
-          if (conversion && conversion.content && conversion.content.data && conversion.content['mime-type']) {
+          if (
+            conversion &&
+            conversion.content &&
+            conversion.content.data &&
+            conversion.content['mime-type'].match(/video.*/)
+          ) {
             conversions.push({
               data: conversion.content.data,
               type: conversion.content['mime-type'],
```

Comment:
Since we're going through an array and conditionally adding elements to another array and returning it in the end, we could also have this as just a return of the original array filtered and then mapped. Something like:
```
        return this.document.properties['vid:transcodedVideos'].filter((conversion) => 
          conversion &&
          conversion.content &&
          conversion.content.data &&
          conversion.content['mime-type'].match(/video.*/)
        ).map((conversion) => {
          data: conversion.content.data,
          type: conversion.content['mime-type'],
        });
```
Replacing lines 284 to 298.
(GitHub is not letting me suggest changes outside your changes)

Reply:
True, I also thought about that but didn't want to introduce that change. For me it makes sense to do it though, it simplifies the code and makes it more readable. Just one detail, on the map function, since we want to return an object inside an array we need also to surround the object structure with parenthesis, in the implicit return. Something like this:
`map((conversion) => ({
     data: conversion.content.data,
     type: conversion.content['mime-type']
}));`

Reply:
Actually, linter complains about implicit return here 😓 , so the proper format should be this:
`map((conversion) => { return { data: conversion.content.data, type: conversion.content['mime-type'],}},);`